scripts/cook_soup.py

```diff
@@ -0,0 +1,407 @@
+}
+
+
+def get_license_object(license_id: str) -> License:
+    """
+    Retrieve the License instance for a given license identifier.
+    If not found, default to the 'unknown' license object.
+
+    :param license_id: e.g. "mit", "apache-2.0", "unknown"
+    :return: License instance
+    """
+    normalized = license_id.lower().strip()
+    return LICENSE_MAPPING.get(normalized, LICENSE_MAPPING["unknown"])
+
+def normalize_license(license_str: str) -> str:
+    """
+    Normalize a license string to lowercase. Fallback to 'unknown' if empty.
+    """
+    if not license_str:
+        return "unknown"
+    return license_str.lower().strip()
+
+
+def process_poetry_licenses(file_path):
+    """
+    Read and parse the JSON output from `pip-licenses` for Poetry-based Python projects.
+    
+    :param file_path: Path to `licenses-poetry.json`.
+    :return: List of dicts with keys: name, version, url, license.
+    """
+    with open(file_path, "r", encoding="utf-8") as f:
+        data = json.load(f)
+
+    results = []
+    for item in data:
+        normalized = normalize_license(item.get("License", ""))
+        results.append({
+            "name": item.get("Name", "Unknown"),
+            "version": item.get("Version", "Unknown"),
+            "url": item.get("URL", ""),
+            "license": normalized
+        })
+    return results
+
+def process_npm_licenses(file_path):
+    """
+    Read and parse the JSON output from `license-checker` for npm-based projects.
+    
+    :param file_path: Path to `licenses-npm.json`.
+    :return: List of dicts, each with keys: name, version, url, license.
+    """
+    with open(file_path, "r", encoding="utf-8") as f:
+        data = json.load(f)
+
+    results = []
+    for package_key, info in data.items():
+        # package_key often looks like "react@17.0.2"
+        if "@" in package_key:
+            name, version = package_key.rsplit("@", 1)
+        else:
+            name = package_key
+            version = "unknown"
+
+        license_str = info.get("licenses", "")
+        if isinstance(license_str, list) and license_str:
+            license_str = license_str[0]
+
+        normalized = normalize_license(license_str)
+        url = info.get("repository") or info.get("url") or ""
+        results.append({
+            "name": name,
+            "version": version,
+            "url": url,
+            "license": normalized
+        })
+    return results
+
+
+def parse_maintainer_from_url(repo_url):
+    """
+    Parse the maintainer (owner) from a GitHub repository URL.
+    E.g. 'https://github.com/apache/commons-lang' => 'apache'
+    
+    If it doesn't match the standard pattern, returns ''.
+    """
+    pattern = r"(?:https?:\/\/)?(?:www\.)?github\.com\/([^\/]+)\/([^\/]+)"
+    match = re.search(pattern, repo_url)
+    if match:
+        owner = match.group(1)
+        return owner
+    return ""
+
+
+def fetch_github_repo_info(owner, repo):
```

Comment:
Do we know how long this function takes to run on average? Is it worth saving time by doing all the API requests simultaneously?

Reply:
Naw, it runs quite quickly. Seconds rather than minutes. And this won't be happening very often.

I wanted to solve this without any dependencies because it makes it easier later on not to have to have dependency management just for the workflow. Generally, I prefer all workflows to just use standard library tools. Thus, no requests or aiohttp here.